src/components/pages/maintenance/maintenance.js

```diff
@@ -110,6 +112,7 @@ class MaintenancePage extends Component {
 
   updateAlarm = (Status) => {
     const { selectedAlarms } = this.state;
+    this.setState({ showIndicator: true });
     Promise.all(
```

Comment:
```
Observable.from(selectedAlarms)
    .do(_ => this.setState({ showIndicator: true }))
    .map(alarm => ({...alarm, Status})))
    .flatMap(ApiService.updateAlarmsStatus)
    .reduce((acc, curr) => [...acc, curr], [])
    .zip(Observable.of(undefined).delay(2000), updatedAlarms => updatedAlarms)
   .subscribe(updatedAlarms => { ...// handle success })
```

By using the zip, of the calls are faster than 2 seconds that is max time the spinner will appear. If the calls take longer than that, the spinner will appear for the length of the call.